clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java

```diff
@@ -522,7 +522,12 @@ private long waitOnMetadata(String topic, long maxWaitMs) throws InterruptedExce
             log.trace("Requesting metadata update for topic {}.", topic);
             int version = metadata.requestUpdate();
             sender.wakeup();
-            metadata.awaitUpdate(version, remainingWaitMs);
+            try {
+                metadata.awaitUpdate(version, remainingWaitMs);
```

Comment:
That could be a useful change. It is a public method though and I am not sure its valuable enough of a change to deal with any compatibility concerns.